openmls/src/group/mls_group/staged_commit.rs

```diff
@@ -243,52 +241,35 @@ impl MlsGroup {
             .confirmation_key()
             .tag(backend, &confirmed_transcript_hash)?;
         if &own_confirmation_tag != received_confirmation_tag {
-            // FIXME: reset nodes. This should get fixed with the tree rewrite.
-            provisional_tree.nodes = original_nodes;
             log::error!("Confirmation tag mismatch");
             log_crypto!(trace, "  Got:      {:x?}", received_confirmation_tag);
             log_crypto!(trace, "  Expected: {:x?}", own_confirmation_tag);
             return Err(StageCommitError::ConfirmationTagMismatch.into());
         }
 
-        // Verify KeyPackage extensions
-        if let Some(path) = &commit.path {
-            if !is_own_commit {
-                let parent_hash = provisional_tree.set_parent_hashes(backend, sender_index)?;
-                if let Some(received_parent_hash) = path
-                    .leaf_key_package
-                    .extension_with_type(ExtensionType::ParentHash)
-                {
-                    let parent_hash_extension =
-                        match received_parent_hash.as_parent_hash_extension() {
-                            Ok(phe) => phe,
-                            Err(_) => return Err(StageCommitError::NoParentHashExtension.into()),
-                        };
-                    if parent_hash != parent_hash_extension.parent_hash() {
-                        return Err(StageCommitError::ParentHashMismatch.into());
-                    }
-                } else {
-                    return Err(StageCommitError::NoParentHashExtension.into());
-                }
-            }
-        }
-
         // If there is a key package from the Commit's update path, add it to the proposal queue
         // TODO #424: This won't be necessary anymore, we can just apply the proposals first
         // and add a new fake Update proposal to the queue after that
         if let Some(key_package) = path_key_package {
             let proposal = Proposal::Update(UpdateProposal { key_package });
-            let staged_proposal =
-                StagedProposal::from_proposal_and_sender(ciphersuite, backend, proposal, sender)
-                    .map_err(|_| MlsGroupError::LibraryError)?;
+            let staged_proposal = StagedProposal::from_proposal_and_sender(
```

Comment:
#424 should be addressed with the tree rewrite as well.

Reply:
With the tree rewrite, path processing implies replacing the current leaf node of the committer with the one in the path. Do I understand correctly, that we can then remove this whole `if let ...`?

Reply:
Let's tackle #424 in a follow-up PR